database/database.js

```diff
@@ -31,18 +31,24 @@ const customerQuestionsSchema = new mongoose.Schema({
 const CustomerQuestions = mongoose.model('CustomerQuestions', customerQuestionsSchema);
 
 const load = (productId, callback) => {
-  CustomerQuestions.find({ productId: productId })
+  CustomerQuestions
+    .aggregate([
+      { $match: { productId: productId }},
+      { $unwind: '$questionAndAnswers' },
+      { $sort: { 'questionAndAnswers.rating': -1 }},
+      { $group: { _id: '$_id',
+        productId: { $first: '$productId'},
+        questionAndAnswers: { $push: '$questionAndAnswers'}}
+      }
+    ])
     .exec((err, data) => {
       // if error, return error
       if (err) {
-        console.log(`Error loading product ${productId} from database`, err);
         callback(err);
       }
       // else if product does not exist, create and send error
-      // db.collections.find() does not return error when no query match
       else if (data[0] === undefined || !data[0].productId) {
-        console.log(`Error: product ${productId} does not exist`);
-        callback(new Error('Product not found!'));
+        callback(new Error(`Product ${productId} not found!`));
```

Comment:
Just curious. You already are using promise returning from exec, any reason for using callback instead of promise chaining ?

Reply:
I am using a callback to handle all errors on server. That way when server receives error from database, it can send error response to the client, rather than having an error end at the database level. That's my reasoning; I hope it makes sense! Please let me know if there is an alternative approach.

Reply:
Your approach makes sense to me.  I was told by SEIR not to mix callbacks in Promise to avoid code getting more complex so I was thinking something like below but need to try it out. Also check this article : https://eddywashere.com/blog/switching-out-callbacks-with-promises-in-mongoose

`CustomerQuestions
    .aggregate([
      { $match: { productId: productId }},
     ....
      }
    ])
    .exec()
  .then(results =>res.json(results)
 .catch(err => res.status(500).json({errormessage:' err...'})`